### PR TITLE
Add bert model qat training sample and cuda gpu optimization and execution sample

### PR DESCRIPTION
**Pull request by zhanghuanrong**

### CODE REVIEW

File: quantization/qat/transformer_qat/ReadMe.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
+# **Bert model QAT and onnxruntime GPU support**
+
+## **Introduction**
+
+This bert_imdb_finetune_qat example is to show how to:
+* Do QAT training with bert model
+* Export QAT trainned model to onnx model with Q/DQ operators
+* optimized the exported Q/DQ model to run with onnxruntime CUDA execution engine in int8 mode.
```
Comment (hariharans29) [inline quantization/qat/transformer_qat/ReadMe.md:8]: optimized -> Optimize ?

execution engine -> execution provider ?

File: quantization/qat/transformer_qat/ReadMe.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+The base fp32 part of the sample is based on following document on web by Fabio Chiusano: 
+[BERT Finetuning with Hugging Face and Training Visualizations with TensorBoard](https://medium.com/nlplanet/bert-finetuning-with-hugging-face-and-training-visualizations-with-tensorboard-46368a57fc97)
+
+## **Prerequest**
```
Comment (hariharans29) [inline quantization/qat/transformer_qat/ReadMe.md:13]: Pre-requisite ?

File: quantization/qat/transformer_qat/ReadMe.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+Following command will use the fp32 model finetuned above, quantize it, and do quantize awared training.
+Evalute the QAT model, then export onnx model with Q/DQ.
+
+```console
+python bert_imdb_finetune_qat.py --do_qat_fine_tune --do_qat_eval --do_qat_export --do_qat_onnx_eval
+```
+
+## **optimize the qdq model for onnxruntime cuda**
+
+Newest onnxruntime contains transformers optimizer for int8 Q/DQ model. To fuse the qdq onnx model, run:
```
Comment (hariharans29) [inline quantization/qat/transformer_qat/ReadMe.md:41]: Newest onnxruntime contains ...-> Starting onnxruntime 1.13, the package contains ... ?

File: quantization/qat/transformer_qat/ReadMe.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+To run and evalute the optimized int8 onnxruntime model,
+```console
+python bert_imdb_finetune_qat.py --do_qat_onnx_eval --qat_opt_onnx_model ../BertModel/bert-base-uncased-finetuned-imdb-int8.onnx
+```
+as you may encounter conflict as torch and onnxruntime here require different cuda, we suggest you creat seperate environment that contains torch cpu with onnxruntime-gpu to run the last command above.
+
+## **How to QAT a Bert Model**
+
+Basically, all QAT change is in the file qat_bert.py and qat_utils.py, following things are done there:
```
Comment (hariharans29) [inline quantization/qat/transformer_qat/ReadMe.md:54]: How about going with "All QAT componesnts are in the files qat_file.py and qat_utils.py" ?

File: quantization/qat/transformer_qat/qat_bert.py [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+        self.quantize_attention_scores = quantize_attention_scores
+        self.quantize_attention_probs = quantize_attention_probs
+        self.quantize_context_layer = quantize_context_layer
+
+
+# could export out for user to control detail
+bert_qat_option = BertQatOptions(quantize_attention_scores = True, quantize_attention_probs = True)
+
+
+class QAwaredBertSelfAttention(nn.Module):
```
Comment (hariharans29) [inline quantization/qat/transformer_qat/qat_bert.py:35]: Awared -> Aware ?

Comment (zhanghuanrong) [reply quantization/qat/transformer_qat/qat_bert.py:35]: Good suggestion.

**Pull request merged**